src/processScript/transform.ts

```diff
@@ -799,15 +799,16 @@ export function transform(
 
 			if (referencePath.parentPath.parentPath.parentPath?.type == `CallExpression`) {
 				// BUG this is causing typescript to be slow
-				referencePath.parentPath.parentPath.replaceWith(t.identifier(`$${uniqueId}$SUBSCRIPT$${
+				referencePath.parentPath.parentPath.replaceWith(t.identifier(`$${uniqueId}$${secLevel}$SUBSCRIPT$${
 					referencePath.parent.property.name
 				}$${referencePath.parentPath.parentPath.node.property.name}$`))
 			} else {
 				const name =
 					`${referencePath.parent.property.name}$${referencePath.parentPath.parentPath.node.property.name}`
 
 				referencePath.parentPath.parentPath.replaceWith(t.identifier(`_${uniqueId}_SUBSCRIPT_${name}_`))
-				neededSubscriptLets.add(name)
+				const maxSecLevel = Math.max(neededSubscriptLets.get(name) || 0, secLevel)
```

Comment:
I'm not sure how I feel about using the highest used seclevel for all subscript calls. You say the use case of someone legitimately making use of varying seclevels for the same subscript is a bit far-fetched, but how much much harder would it be to be correct and support that use case?

Reply:
Okay I've thought about it, and I think I've decided I'm happy with this behaviour as it is.